### PR TITLE
Port SDS to V3 api

### PR DESCRIPTION
**Pull request by tvh**
This addresses https://github.com/smallstep/step-sds/issues/7

### CODE REVIEW

File: sds/service.go [hunk 1]
```diff
@@ -84,15 +86,19 @@ func (srv *Service) Stop() error {
 
 // Register registers the sds.Service into the given gRPC server.
 func (srv *Service) Register(s *grpc.Server) {
-	discovery.RegisterSecretDiscoveryServiceServer(s, srv)
+	secret.RegisterSecretDiscoveryServiceServer(s, srv)
+}
+
+func (srv *Service) DeltaSecrets(sds secret.SecretDiscoveryService_DeltaSecretsServer) (err error) {
+	return errors.New("DeltaSecretsServer not implemented")
```
Comment (tvh) [inline sds/service.go:93]: This works if the `api_type` is set to `GRPC`. See https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/config_source.proto.html?highlight=api_type#envoy-v3-api-enum-config-core-v3-apiconfigsource-apitype.

**Review by maraino — COMMENTED** [review]
Thanks @tvh,
LGTM. I've also changed the envoy-sds server in the example that uses UNIX sockets, it wasn't working before.

**Comment by tvh** [comment]
> I've also changed the envoy-sds server in the example that uses UNIX sockets, it wasn't working before.

Yes thank you, I neglected to do that.

**Comment by tvh** [comment]
@maraino Would you mind publishing a new docker image as well? We are currently using a fork in our own registry.

**Comment by maraino** [comment]
I'm trying, I've just fixed some linter errors.

And I'm not sure about the status of the CI right now, we moved most of our repos to GH actions, but this is still using Travis.

**Comment by maraino** [comment]
@tvh even though the build failed in the last step, the docker images seem to be up

**Pull request merged**